poetry/packages/project_package.py

```diff
@@ -8,6 +8,23 @@
 
 class ProjectPackage(Package):
     def __init__(self, name, version, pretty_version=None):
+        if isinstance(version, str) and version.startswith('attr:'):
+            version_pkg, _, version_attr = (
+                version[6:].strip().rpartition('.')
```

Comment:
Should `"attr:function_name"` be allowed, without a space after the colon?

Reply:
Why not? It's harmless and it's how it works in setuptools.